src/libraries/System.Runtime/tests/System/DoubleTests.cs

```diff
@@ -353,8 +353,15 @@ internal static string SplitPairs(string input)
             string ret = "";
             foreach (var pair in splitPairs)
             {
-                string reversedPair = Reverse(pair);
-                ret += reversedPair;
+                if (BitConverter.IsLittleEndian)
+                {
+                    string reversedPair = Reverse(pair);
+                    ret += reversedPair;
+                }
+                else
+                {
+                    ret += pair;
+                }
```

Comment:
Nit:
```C#
if (BitConverter.IsLittleEndian)
{
    return input.Remove("-");
}

return string.Concat(input.Split("-").Select(pair => Reverse(pair)));
```

Reply:
That's indeed more elegant; patch updated.   (I've also had to add a "using System.Linq" statement.)